### PR TITLE
Package camlpdf.2.7.2

### PR DESCRIPTION
**Pull request by johnwhitington**
### `camlpdf.2.7.2`
Read, write and modify PDF files

---
* Homepage: http://github.com/johnwhitington/camlpdf
* Source repo: git://github.com/johnwhitington/camlpdf
* Bug tracker: http://github.com/johnwhitington/camlpdf/issues

---
:camel: Pull-request generated by opam-publish v2.4.0

### CODE REVIEW

**Comment by shonfeder** [comment]
The only CI failure is on windows (https://github.com/ocaml/opam-repository/actions/runs/11543903013/job/32128826435?pr=26794)

```
  <><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
  -> retrieved camlpdf.2.7.2  (https://github.com/johnwhitington/camlpdf/archive/refs/tags/v2.7.1.tar.gz)
  -> retrieved ocamlfind.1.9.5  (http://download.camlcity.org/download/findlib-1.9.5.tar.gz)
  [WARNING] .install file is missing .exe extension for src/findlib/ocamlfind
  [WARNING] .install file is missing .exe extension for src/findlib/ocamlfind_opt
  [WARNING] Automatically adding .exe to D:\opamroot\default\.opam-switch\build\ocamlfind.1.9.5\src\findlib\ocamlfind.exe
  [WARNING] Automatically adding .exe to D:\opamroot\default\.opam-switch\build\ocamlfind.1.9.5\src\findlib\ocamlfind_opt.exe
  [WARNING] D:\opamroot\default\bin\safe_camlp4 is a script; the command won't be available
  -> installed ocamlfind.1.9.5
  [ERROR] The compilation of camlpdf.2.7.2 failed at "make".
  
  #=== ERROR while compiling camlpdf.2.7.2 ======================================#
  # context     2.2.1 | win32/x86_64 | ocaml.5.2.0 | file://D:/a/opam-repository/opam-repository
  # path        D:\opamroot\default\.opam-switch\build\camlpdf.2.7.2
  # command     D:\opamroot\.cygwin\root\bin\make.exe
  # exit-code   2
  # env-file    D:\opamroot\log\camlpdf-3676-0aaf04.env
  # output-file D:\opamroot\log\camlpdf-3676-0aaf04.out
  ### output ###
  # [...]
  # OCamlMakefile:1163: warning: pattern recipe did not update peer target '._ncdi/pdfocg.di'.
  # ocamldep pdfmerge.mli > ._bcdi/pdfmerge.di
  # OCamlMakefile:1163: warning: pattern recipe did not update peer target '._ncdi/pdfmerge.di'.
  # ocamlc   \
  # 			-c -cc "cc"     -ccopt "-fPIC -o2 -D__MINGW32__ -D__MINGW32__ \
  # 			-DPIC   \
  # 			    -o flatestubs.o " flatestubs.c
  # 'cc' is not recognized as an internal or external command,
  # operable program or batch file.
  # make[1]: *** [OCamlMakefile:1132: flatestubs.o] Error 2
  # make[1]: Leaving directory '/cygdrive/d/opamroot/default/.opam-switch/build/camlpdf.2.7.2'
  # make: *** [OCamlMakefile:792: byte-code-library] Error 2
  
  
  
  <><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
  +- The following actions failed
  | - build camlpdf 2.7.2
  +- 
  +- The following changes have been performed
  | - install ocamlfind 1.9.5
  +- 
  # Run (& opam env) -split '\r?\n' | ForEach-Object { Invoke-Expression $_ } to update the current shell environment
  
  The former state can be restored with:
      D:\opam\bin\opam.exe switch import "D:\\opamroot\\default\\.opam-switch\\backup\\state-20241027212234.export"
camlpdf.2.7.2 failed to build.
```

Is the build meant to work on windows?

**Comment by johnwhitington** [comment]
Well, it builds on Windows here. But then, I don't use OPAM on Windows, and am generally not up to date on modern best practices for OCaml on Windows.

Having said that, having the environment not recognise "cc" as a C compiler seems odd... Don't we normally symlink "cc" to the current C compiler?

File: packages/camlpdf/camlpdf.2.7.2/opam [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+bug-reports: "http://github.com/johnwhitington/camlpdf/issues"
+dev-repo: "git://github.com/johnwhitington/camlpdf"
+license: "LGPL-2.1-or-later WITH OCaml-LGPL-linking-exception"
+build: [[make]]
+install: [[make "install"]]
+depends: [
+  "ocaml" {>= "4.10"}
+  "ocamlfind" {build}
+]
+synopsis: "Read, write and modify PDF files"
```
[the remark this answers is not in the document]
Comment (mseri) [reply packages/camlpdf/camlpdf.2.7.2/opam:14]: Let's see if it is just this or the full native one that is not compiling

**Comment by mseri** [comment]
I think it depends... this probably works fine with cygwin. You can see the compilcations with windows at the bottom of this page https://github.com/ocaml/opam-repository/wiki/Depexts-os-distribution---os-family-values

**Pull request merged**